doc/authoring_command_modules/authoring_commands.md

```diff
@@ -77,108 +95,240 @@ with self.command_group('mymod', mymod_sdk) as g:
+  c.argument('my_param', ...)
 ```
-- `arg_type` - An instance of the `azure.cli.core.commands.CliArgumentType` class. This essentially serves as a named, reusable packaging of the `kwargs` that modify your command's argument. It is useful when you want to reuse an argument definition, but is generally not required. It is most commonly used for name type parameters.
-- `kwargs` - Most likely, you will simply specify keyword arguments in `register_cli_argument` that will accomplish what you need.  Any `kwargs` specified will override or extended the definition in `arg_type`, if provided.
+- `kwargs` - Any supported kwarg which will be applied to calls within the context block.
+
+**(2) Register different kinds of arguments**
+
+***argument***
+```Python
+argument(self, dest, arg_type=None, **kwargs)
+```
+- `dest` -  The name of the parameter that you are targeting.
+- `arg_type` - An instance of the `azure.cli.core.commands.CliArgumentType` class. This essentially serves as a named, reusable packaging of the `kwargs` that modify your command's argument. It is useful when you want to reuse an argument definition, but is generally not required. It is most commonly used for name type parameters, or for enums and flags.
+- `kwargs` - Most likely, you will simply specify keyword arguments that will accomplish what you need.  Any `kwargs` specified will override or extended the definition in `arg_type`, if provided.
+
+***ignore***
+```Python
+ignore(self, *args)
+```
+- `args` - one or more parameter names (dest values) that should be ignored. Useful to suppress arguments that appear due to reflection on an SDK, but are unwanted in the command signature.
+
+***extra***
+```Python
+extra(self, dest, arg_type=None, **kwargs)
+```
+Arguments are the same as `argument`, however this will create a new parameter whereas `argument` will not. This is useful when a reflected SDK method is missing a parameter that you need to expose in your command.
+
+***expand***
+```Python
+expand(self, dest, model_type, group_name=None, patches=None):
+```
+
+Often reflected SDK methods have complex parameters that are difficult to expose directly. The `expand` method offers one way to expose these methods without resorting to a custom command approach.
+
+- `dest` -  The name of the parameter that will be expanded.
+- `model_type` - The model type which will be expanded and collapsed back into the `dest` value.
+- `group_name` - The argument group to which the expand parameters will be assigned. (See arg_group kwarg)
+- `patches` - A list of patches to apply to the expanded parameters.
+
+Additional Topics
+=============================
+
+## Keyword Argument Reference
+
+**Overview of Keyword Arguments in Azure CLI 2.0**
+
+When writing commands for Azure CLI 2.0, it is important to understand how keyword arguments (kwargs) are applied. Refer to the following diagram.
+
+![](/doc/assets/annotated-kwarg-structure.gif)
```

Comment:
nit: Remove the red underline under 'kwargs' from the image if it's easy enough to do.

Reply:
Oh, haha, that's what happens when you take a screenshot from a powerpoint with spell-check enabled. :rofl: